Source/radiation/MGFLD.cpp

```diff
@@ -1001,16 +1002,1238 @@ void Radiation::compute_limiter(int level, const BoxArray& grids,
     
     const Real* dx = parent->Geom(level).CellSize();
 
+    using namespace filter;
+
 #ifdef _OPENMP
 #pragma omp parallel
 #endif    
     for (MFIter mfi(Er_wide,false); mfi.isValid(); ++mfi) {
-      BL_FORT_PROC_CALL(CA_COMPUTE_LAMBORDER, ca_compute_lamborder)
-        (BL_TO_FORTRAN(Er_wide[mfi]), 
-         BL_TO_FORTRAN(kpr[mfi]),
-         BL_TO_FORTRAN(lamborder[mfi]), 
-         dx, &ngrow, &radiation::limiter, &filter_lambda_T, &filter_lambda_S);
-    }
+        FArrayBox lamfilfab;
+        Box lambx = lamborder[mfi].box();
+        Box bx = grow(lambx, -ngrow);
+
+        if (filter_lambda_T > 0) {
+            lamfilfab.resize(bx);
+        }
+
+        Array4<Real> const lam = lamborder[mfi].array();
+        Array4<Real> const Er = Er_wide[mfi].array();
+        Array4<Real> const kap = kpr[mfi].array();
+        Array4<Real> const lamfil = lamfilfab.array();
+
+        for (int g = 0; g < Radiation::nGroups; ++g) {
+
+            amrex::Loop(lambx, [=] (int i, int j, int k) noexcept
+            {
+                lam(i,j,k,g) = -1.e50_rt;
+
+                if (Er(i,j,k,g) == -1.e0_rt) {
+                    return;
+                }
+
+                Real r1 = 0.0;
+                Real r2 = 0.0;
+                Real r3 = 0.0;
+
+                if (Er(i-1,j,k,g) == -1.e0_rt) {
```

Comment:
are we sure this is correct?  The original code is just looking at the boundary of the box, and the expectation is that the boundary is filled with -1.  But now you are looking at the entire box, and checking if things are -1.  I presume that Er should always be >= 0, so this still works?

maybe we should add a comment explaining that -1 marks the boundary?

Reply:
The 1D and 2D/3D code approached this differently and I had to decide which pattern to go with. I arbitrarily chose the 3D version just because that's the one I started the port with. In either case you do have to do an operation for every zone, but the 1D code handles the interior in one step and the boundary in a second step, while this code handles both in one loop. If you extended the 1D approach for 3D then you'd have six separate boundary loops which just seems unnecessary, though maybe marginally more computationally efficient.